### PR TITLE
Feature/integrate-ag-grid-full-width

### PR DESCRIPTION
**Pull request by J-DK**
# Description

Integrated Ag-grid full width row.

Fixes # (add-new-feature)

## Type of change

Please check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/app/routes/features/features.component.ts [hunk 1]
```diff
@@ -12,13 +15,29 @@ import MapUtils from '../../shared/utils/map.utils';
...
+  gridApi: GridApi;
+  columnApi: ColumnApi;
+  gridOptions: GridOptions;
+  rowData: Feature[];
 
   constructor(private featureService: FeatureService, private logger: NGXLogger) {
+      this.gridOptions = {
+          headerHeight: 0,
+          columnDefs: [],
+          rowHeight: 160,
```
Comment (paul58914080) [inline src/app/routes/features/features.component.ts]: Shouldn't the rowHeight be auto-adjustable ?

File: src/app/routes/features/features.component.ts [hunk 2]
```diff
@@ -12,13 +15,29 @@ import MapUtils from '../../shared/utils/map.utils';
...
+              fullWidthCellRenderer: FeatureRendererComponent
+          },
+          isFullWidthCell: (rowNode: RowNode) => true
+      };
   }
 
   ngOnInit() {
     this.featureService.getFeatures().subscribe((response) => {
       this.features = response;
+      this.rowData = response;
```
Comment (paul58914080) [inline src/app/routes/features/features.component.ts]: Any reason why we need this.features and this.rowData. Both seem to be the household for the same.

File: src/app/routes/features/features.component.html [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
+         [gridOptions]="gridOptions"
          (gridReady)='onGridReady($event)'>
      </ag-grid-angular>
-     -->
-    <div class='container-fluid'>
-        <div *ngFor="let feature of features">
-            <ff4j-feature-card [feature]="feature"></ff4j-feature-card>
-        </div>
-    </div>
+    <!--<div class='container-fluid'>-->
```
Comment (paul58914080) [inline src/app/routes/features/features.component.html]: Remove unused commented lines

File: src/app/routes/features/features.component.ts [hunk 4]
```diff
@@ -33,4 +52,10 @@ export class FeaturesComponent implements OnInit {
       }
     });
   }
+
+  onGridReady(params) {
+    this.gridApi = params.api;
+    this.columnApi = params.columnApi;
+    this.gridApi.sizeColumnsToFit();
```
Comment (paul58914080) [inline src/app/routes/features/features.component.ts]: Why do we need to `sizeColumnsToFit();`

File: src/app/routes/features/features.component.html (hunk 3 above)
Comment (J-DK) [reply src/app/routes/features/features.component.html]: Ok Paul

File: src/app/routes/features/features.component.ts (hunk 4 above)
Comment (J-DK) [reply src/app/routes/features/features.component.ts]: Yeah, no need of that. Will remove it.

File: src/app/routes/features/features.component.ts (hunk 2 above)
Comment (J-DK) [reply src/app/routes/features/features.component.ts]: It is needed for the commented code. As I ll remove the commented code, I ll refactor this part as well.

**Pull request merged**